### PR TITLE
lint: use sentry-flake8

### PR DESCRIPTION
**Pull request by joshuarli**
Note that we'll need to publish `sentry-plugins` to PyPI before this can be merged (CI needs to pass).

### CODE REVIEW

**Review by mattrobenolt — APPROVED** [review]
🥇

**Review by dcramer — APPROVED** [review]
josh's fault if it breaks

**Comment by JoshuaRLi** [comment]
😭

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -12,6 +12,14 @@ exclude: >
...
     )
 repos:
+-   repo: local
+    hooks:
+    - id: flake8
+      name: flake8
+      entry: flake8
+      language: python
+      types: [python]
+      log_file: '.artifacts/flake8.pycodestyle.log'
```
Comment (asottile) [inline .pre-commit-config.yaml]: note here that you can `additional_dependencies: [sentry-flake8==...]` and then you don't need to depend on it being globally installed

**Pull request merged**